fms/models/llama.py

```diff
@@ -396,10 +396,54 @@ def _rename_weights_to_fms(orig_sd):
     return new_sd
 
 
+def _hf_sd_to_fms_sd(hf_sd: OrderedDict):
+    replacements = [
+        (r"^lm_head.weight", "shared.head.weight"),
+        (r"^model.embed_tokens.weight", "shared.emb.weight"),
+        (r"^model.norm", "dec_norm"),
+        (r"^model.layers", "layers"),
+        (r"self_attn\.k_proj", "attn.key"),
+        (r"self_attn\.v_proj", "attn.value"),
+        (r"self_attn\.q_proj", "attn.query"),
+        (r"self_attn\.o_proj", "attn.dense"),
+        (r"mlp\.gate_proj", "ff_sub_layer.wg"),
+        (r"mlp\.up_proj", "ff_sub_layer.w1"),
+        (r"mlp\.down_proj", "ff_sub_layer.w2"),
+        (r"input_layernorm", "ln"),
+        (r"post_attention_layernorm", "ff_ln"),
+    ]
+    new_sd = {}
+
+    # nheads is used in the transformation required for hf->fms
+    # here we are using 128 as this value fits with all popular models 7B, 13B, 70B to recover the number of heads
+    emb_dim = hf_sd["model.embed_tokens.weight"].size(1)
```

Comment:
what I meant was can we get this inside `if bool(trans_required_pattern.match(new_name)):` by getting the emb_dim from the shape of the k/q? that way we can restore a layer/module from a checkpoint in isolation, even if it didn't contain embed_tokens or any other module.